### PR TITLE
disable some combination of match clause

### PR DESCRIPTION
**Pull request by jievince**

### CODE REVIEW

**Review by czpmango — APPROVED** [review]

File: src/planner/match/SegmentsConnector.cpp [hunk 1]
```diff
@@ -17,13 +17,9 @@ StatusOr<SubPlan> SegmentsConnector::connectSegments(CypherClauseContextBase* le
                                                      SubPlan& left,
                                                      SubPlan& right) {
     UNUSED(rightCtx);
-    if (leftCtx->kind == CypherClauseKind::kReturn) {
```
Comment (Shylock-Hg) [inline src/planner/match/SegmentsConnector.cpp:20]: Plz add test cases.

Comment (jievince) [reply src/planner/match/SegmentsConnector.cpp:20]: OK

File: src/context/ast/QueryAstContext.h [hunk 2]
```diff
@@ -27,6 +27,8 @@ enum class CypherClauseKind : uint8_t {
     kYield,
 };
 
+const char* magic(CypherClauseKind e);
```
Comment (czpmango) [inline src/context/ast/QueryAstContext.h:30]: Is this a kind of magic？

File: src/planner/match/SegmentsConnector.cpp [hunk 3]
```diff
@@ -17,21 +17,19 @@ StatusOr<SubPlan> SegmentsConnector::connectSegments(CypherClauseContextBase* le
...
         VLOG(1) << "left tail: " << left.tail->outputVar()
                 << "right root: " << right.root->outputVar();
         addInput(left.tail, right.root);
         left.tail = right.tail;
         return left;
     }
 
-    return Status::Error("Can not solve the connect strategy of the two subplan..");
+    return Status::Error("Can not solve the connect strategy of the two subplans %s and %s",
+                         magic(rightCtx->kind),
```
Comment (czpmango) [inline src/planner/match/SegmentsConnector.cpp:31]: Can these codes work?
```cpp
     std::stringstream ss;
     ss << "some error msg: "<<rightCtx->kind;
     return Status::Error(ss.str());
```

File: src/context/ast/QueryAstContext.h (hunk 2 above)
Comment (jievince) [reply src/context/ast/QueryAstContext.h]: has deleted

**Review by yixinglu — APPROVED** [review]

**Review by Aiee — APPROVED** [review]

**Pull request merged**